### PR TITLE
Typescript optimize

### PR DESCRIPTION
**Pull request by youluna**

### CODE REVIEW

File: src/checkbox/checkbox-group.jsx [hunk 1]
```diff
@@ -27,7 +27,10 @@ class CheckboxGroup extends Component {
         /**
          * 可选项列表, 数据项可为 String 或者 Object, 如 `['apple', 'pear', 'orange']` 或者 `[{value: 'apple', label: '苹果',}, {value: 'pear', label: '梨'}, {value: 'orange', label: '橙子'}]`
          */
-        dataSource: PropTypes.arrayOf(PropTypes.any),
+        dataSource: PropTypes.oneOfType([
```
Comment (guanpu) [inline src/checkbox/checkbox-group.jsx]: This change actually exclude 'function' from datasource option, I'm afraid there is potential issue on this change

File: src/radio/radio-group.jsx [hunk 2]
```diff
@@ -73,7 +73,10 @@ class RadioGroup extends Component {
         /**
          * 可选项列表, 数据项可为 String 或者 Object, 如 `['apple', 'pear', 'orange']`
          */
-        dataSource: PropTypes.arrayOf(PropTypes.any),
+        dataSource: PropTypes.oneOfType([
```
Comment (guanpu) [inline src/radio/radio-group.jsx]: It has the same potential problem as commented in checkbox-group

File: src/checkbox/checkbox-group.jsx (hunk 1 above)
Comment (youluna) [reply src/checkbox/checkbox-group.jsx]: you mean array of functions?

**Comment by youluna** [comment]
Radio/Chekbox 's dataSource type should be the same, and this is before: 
```js
// ...jsx
 dataSource: PropTypes.arrayOf(PropTypes.any),

// ...ts
dataSource?: Array<any> 
```
this is after:

```js
// ...jsx
dataSource: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.string),
            PropTypes.arrayOf(PropTypes.object),
]),

// ...ts
dataSource?: Array<string> | Array<data>
```
is this ok?

File: src/checkbox/checkbox-group.jsx (hunk 1 above)
Comment (guanpu) [reply src/checkbox/checkbox-group.jsx]: My bad, never mind this.

**Comment by guanpu** [comment]
> Radio/Chekbox 's dataSource type should be the same, and this is before:
> 
> ```js
> // ...jsx
>  dataSource: PropTypes.arrayOf(PropTypes.any),
> 
> // ...ts
> dataSource?: Array<any> 
> ```
> 
> this is after:
> 
> ```js
> // ...jsx
> dataSource: PropTypes.oneOfType([
>             PropTypes.arrayOf(PropTypes.string),
>             PropTypes.arrayOf(PropTypes.object),
> ]),
> 
> // ...ts
> dataSource?: Array<string> | Array<data>
> ```
> 
> is this ok?

I prefer not to narrow previous allowed types. I'll let @myronliu347 @bindoon judge

**Pull request merged**